OpenTrace/Components/StyledButton.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  StyledButton.swift
+//  OpenTrace
+//
+//  Created by Robbie Fraser on 15/04/2020.
+//  Copyright © 2020 OpenTrace. All rights reserved.
+//
+
+import UIKit
+
+final class StyledButton: UIButton {
+
+	var appearance: Appearance = .primaryFill {
+		didSet {
+			setup()
+		}
+	}
+
+	private var borderLayer = CAShapeLayer()
+
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+		setup()
+	}
+
+	required init?(coder: NSCoder) {
+		super.init(coder: coder)
+		setup()
+	}
+
+	override func layoutSubviews() {
+		super.layoutSubviews()
+		updateBorederLayerPath()
+	}
+
+	private func setup() {
+		cornerRadius = 8
+		setupAppearance()
+		setupBorderLayer()
+		heightAnchor.constraint(equalToConstant: 56).isActive = true
+		titleLabel?.font = UIFont(descriptor: .preferredFontDescriptor(withTextStyle: .headline), size: 16)
+	}
+
+	private func setupAppearance() {
+		backgroundColor = appearance.backgroundColour
+		borderColor = appearance.borderColor
+		setTitleColor(appearance.textColour, for: .normal)
+	}
+
+	private func setupBorderLayer() {
```

Comment:
Why does the border need to be an extra layer? can we not just use `self.layer.borderWidth = 1`? then we never need to worry about resizing etc

Reply:
Yeh much simpler. 
Will update 👍